### PR TITLE
net/linux: add net_netlink_addrs

### PR DESCRIPTION
**Pull request by sreimers**
This pull request adds support for obtaining network interface addresses using the RTNETLINK API on Linux systems.

fixes https://github.com/baresip/baresip/issues/3245

### CODE REVIEW

**Comment by alfredh** [comment]
seems to work on Linux:

```
Local network addresses:
     enp2s0:  10.0.1.2
     enp2s0:  fe80::ea9c:25ff:fe8a:f1fe
```

**Comment by sreimers** [comment]
@cspiel1 @maximilianfridrich @juha-h would be nice if you can test this PR too.

**Comment by juha-h** [comment]
I built baresip app using re of this PR and baresip app on Debian worked OK over WiFi interface.

File: src/net/net.c [hunk 1]
```diff
@@ -129,6 +129,9 @@ int net_default_source_addr_get(int af, struct sa *ip)
  */
 int net_if_apply(net_ifaddr_h *ifh, void *arg)
 {
+#ifdef LINUX
+	return net_netlink_addrs(ifh, arg);
+#endif
```
Comment (maximilianfridrich) [inline src/net/net.c:134]: Would a `#elif` be cleaner here?

**Comment by maximilianfridrich** [comment]
I just tested it with my normal setup and with a setup mimicking @robert-scheck's. Looks good.

PS: I setup a new bridge interface `br9` and assigned a tentative IPv6 using `sudo ip addr add fd00::1/64 dev br9 tentative`. With main branch, it crashed, with this branch, the tentative IPv6 is ignored.

**Review by robert-scheck — APPROVED** [review]
I can't comment about the actual code changes, just about the result: Yes, it solves the problem, thank you :)

**Comment by sreimers** [comment]
Thanks all for testing and review!

**Pull request merged**